pkg/controllers/operator/capture/controller.go

```diff
@@ -385,18 +386,23 @@ func (cr *CaptureReconciler) handleDelete(ctx context.Context, capture *retinav1
 	}
 	cr.logger.Info("Capture jobs are removed", zap.String("Capture", captureRef.String()))
 
+	// Remove the secret if the secret is created by the operator when the managed storage account is enabled.
 	if cr.managedStorageAccountEnabled() {
-		secret := getSecretFromCapture(capture, "")
-
-		if err := apiretry.Do(
-			func() error {
-				return cr.Client.Delete(ctx, &secret) //nolint:wrapcheck // no wrapped, detailed explanation is required for the internal error
-			},
-		); err != nil {
-			cr.logger.Error("Failed to delete secret", zap.Error(err), zap.String("Capture", captureRef.String()))
-			return ctrl.Result{}, fmt.Errorf("failed to delete secret: %w", err)
+		managedSecret := getSecretFromCapture(capture, "")
+		// Delete the secret only when the secret is created by the operator.
+		if *capture.Spec.OutputConfiguration.BlobUpload == managedSecret.Name {
```

Comment:
risk of a nil pointer panic. Check if `BlobUpload` is nil?

Reply:
Good point, checked the nil pointer first.